Goon chem fixes and reorganization

phil235 opened the pull request:
Reorganizing the reagents files. Goon chems are no longer in separate files. Moved grenade launcher and syringe gun from reagents module folder to gun folder.
The reagents module will now look like this:
- Chemistry-Colours.dm
- Chemistry-Holder.dm
- Chemistry-Machinery.dm
- Chemistry-Reagents.dm     <---- no reagents defined there, only procs.
- Chemistry-Recipes.dm        <---- no chemical reactions defined there, only procs.
- reagents_containers.dm
- reagent_dispensers.dm
- Chemistry-Reagents (Folder)
  - Blob, Drug, Medicine, Others, Pyrotechnics, Toxins (all .dm) (the reagents)
  - Consumable-Reagents (Folder)
- Chemistry-Recipes (Folder)
  - Drug, Medicine, Others, Pyrotechnics, Slime_extracts, Toxins (all .dm) (the chem reactions)
- reagent_containers (Folder)

ALSO:
- All reagents are back to being metabolized every tick. I adjusted the goon chem so their effects aren't changed too much.
- Removing some probability and randomness from goon chem effects to make it more predictable.
- Adding overdose to space drug.
- Making a drug reagent category.
- Changed check_ear_prot() for carbons: alien are now immune, and monkey checks their head for ear protection.
- Changed flashpowder flashing code and blackpowder explosion code to use already existing procs.
- Moved chemical_mob_spawn and goon_vortex procs to /datum/chemical_reaction/
- Fixes double overdose message for sugar.
- Remove the "if(!M) M = holder.my_atom" at the beginning of a lot of on_mob_life procs, you already can't call on_mob_life if M is null(Chemistry-Holder.dm).
- Moved random_color_list to datum/reagent/colorful_reagent
- Nerfed healing power of stimulant. Buffed ephedrine a bit.
- Fixes some goon chem on_reaction() proc leaving resulting chem behind when they shouldn't.
- Removes Life and Corgium reagents. They're now only a chemical reaction.
- Added a message for the ghost when someone is trying to revive its corpse with strange reagent. (like defib)
- Fixes Frost oil and other consumable reagents using current_cycle var.
- Fixes runtime where during metabolization a reagent removes another then the latter still calls on_mob_life().
- Fixes antihol only removing pure ethanol instead of all alcoholic reagents.
- Changed antihol recipe (now needs copper) to avoid conflict with sterilizine.
- Fixes broken floor without icon when burning wall with clf3, it now makes a plating instead of a tile. Also, clf3 has a chance to burn platings/tiles if it doesn't destroy it.
- Makes Nitroglycerin stabilizable with stabilizing agent and can explode when heated to 474K.
- Some other tiny fixes.

Fixes #8945.
Fixes #8831.
Fixes #8632
Fixes #8849
Fixes #8500
Fixes #8493 
Fixes #8621
Fixes #7675.

phil235 committed:
Reorganizing the reagents files. Goon chems are no longer in separate files.
All reagents are back to being metabolized every tick. I adjusted the goon chem so their effects aren't changed too much.
Removing some probability and randomness from goon chem effects to make it more predictable.
Adding overdose to space drug.
Making a drug reagent category.
Changed check_ear_prot() for carbons: alien are now immune, and monkey checks their head for ear protection.
Changed flashpowder flashing code and blackpowder explosion code to use already existing procs.
Moved chemical_mob_spawn and goon_vortex procs to /datum/chemical_reaction/
Fixes double overdose message for sugar.
Remove the "if(!M) M = holder.my_atom" at the beginning of a lot of on_mob_life procs, you already can't call on_mob_life if M is null

(Chemistry-Holder.dm).
Moved random_color_list to datum/reagent/colorful_reagent
Nerfed healing power of stimulant. Buffed ephedrine a bit.
Fixes some goon chem on_reaction() proc leaving resulting chem behind when they shouldn't.
Removes Life, nitroglycerin and Corgium reagents. They're now only a chemical reaction.
Moved grenade launcher and syringe gun from reagents module folder to gun folder.
Added a message for the ghost when someone is trying to revive its corpse with strange reagent. (like defib)
Fixes Frost oil and other consumable reagents using current_cycle var.
Fixes runtime where during metabolization a reagent removes another then the latter still calls on_mob_life().

hornygranny commented:
TYBP

Menshin commented:
Great work :+1:

AlexCrimson commented:
Did you make all the reagents in the Chem Dispenser list alphabetically? I feel like that is a super high priority fix.

Fox-McCloud commented:
Goon's chems are meant to be processed every tick---it's just the life proc is meant to take place once every 2 seconds instead of ~1.5-1.8.

If you guys ever port the process scheduler, there won't be need to adjust reagent numbers.

phil235 committed:
Fixes antihol recipe conflicting with sterilizine.
Fixes clf3 making floor with no sprite when burning a wall.
clf3 now burns tiles and platings if it can't destroy them.
Fixes a tiny mistake in consumable/ethanol.

Dorsisdwarf commented:
Nitroglycerine should be an actual reagant because it's a real chemical that can actually be stabilised/

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
+datum/reagent/medicine/calomel/on_mob_life(var/mob/living/M as mob)
+	for(var/datum/reagent/R in M.reagents.reagent_list)
+		if(R != src)
+			M.reagents.remove_reagent(R.id,2.5)
```
Whoa whoa whoa hold the fuck up here

Revert this change! You nerfed calomel into the ground!

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
+	description = "Reduces low radiation damage very effectively."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 2 * REAGENTS_METABOLISM
```
Just get rid of REAGENTS_METABOLISM entirely, please. Unless it's equal to 1.

phil235 replied:
Isn't it used to adjust the overall metabolization speed through config?

Iamgoofball replied:
Last I checked it was a hard define. It's 0.6.

Cheridan replied:
It's not a config, but it was added so that it's easy to adjust metabolization speed. It was originally added as a baystation compatibility update, since their chems metabolize more slowly.

Iamgoofball replied:
Whoever set the bay metabolization rates had to have been insane. Some of them metabolize slower then black powder does.

phil235 replied:
I did not simply multiply by REAGENTS_METABOLISM, the metabolization rates are unchanged, instead of reading "2" it reads "4 * REAGENTS_METABOLISM" for example.

Iamgoofball replied:
What? it says "metabolization_rate = 2 * REAGENTS_METABOLISM" on my end. That's a rate of 0.8.

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
+datum/reagent/medicine/salbutamol/on_mob_life(var/mob/living/M as mob)
+	M.adjustOxyLoss(-3*REM)
+	if(M.losebreath >= 4)
+		M.losebreath -= 2
```
This needs to be -4.

phil235 replied:
It was -4 and being called once every two life(). Now that it's called every life() the effect needs to be halved.

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -152,5 +150,591 @@ datum/reagent/medicine/spaceacillin
+	M.status_flags |= IGNORESLOWDOWN
+	if(cycle_count >= 36)
+		M.sleeping += 3
+	cycle_count++
```
Why did you remove my cycle_count variable from the base reagents thing? Please move var/cycle_count back up to all reagents and make it increase by 1 every time on_mob_life() is called.

phil235 replied:
will do

Iamgoofball commented on code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm:
```diff
@@ -18,13 +57,6 @@
-	color = "#FF0000"
-	metabolization_rate = 4
 
 /datum/chemical_reaction/clf3
```
Why did you separate the recipes from the reagents?

I kept them in the same file as the reagents for easy access to tweaking all things about a reagent without having to flip between files.

phil235 replied:
Because the original chemicals already separated them, plus it makes smaller files.

Iamgoofball commented on code/modules/reagents/Chemistry-Recipes/Pyrotechnics.dm:
```diff
@@ -176,27 +141,12 @@
-	return
+	var/location = get_turf(holder.my_atom)
+	var/datum/effect/effect/system/reagents_explosion/e = new()
+	e.set_up(round(created_volume/8, 1), location, 1, 4, message = 0)
```
Please make sure the explosion scales up to 300 units /exactly/.

phil235 replied:
I tried my best to stick to the numbers in the blackpowder explosion proc.

Iamgoofball replied:
Alright.

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
+	if(M.brainloss + M.toxloss <= 60)
+		M.adjustBrainLoss(1*REM)
+		M.adjustToxLoss(1*REM)
+	if(current_cycle == 54)
```
Might wanna change this to 
if(current cycle >= 54)

phil235 replied:
will do

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
+	reagent_state = LIQUID
+	color = "#CF3600"
+	metabolization_rate = 0.125 * REAGENTS_METABOLISM
+	toxpwr = 0.75
```
Please don't put raw toxpwr on these. I don't want them doing raw toxin damage in some instances.

phil235 replied:
I didn't change anything, cynaide/on_mob_life() already had adjustToxLoss(0.75), I simply moved it to be dealt in toxin/on_mob_life()

Iamgoofball replied:
Oh, okay.

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
+	toxpwr = 2.5
+
+datum/reagent/toxin/initropidril/on_mob_life(var/mob/living/M as mob)
+	if(prob(5))
```
Revert this if(prob(5)) shit.

Revert it on all the chems you added it to. It made all the toxins damn near worthless.

Initro is one of the ones you spend TCs on, why did you nerf it into the ground?

phil235 replied:
I don't understand, didn't you put that prob there? Also I didn't nerf that chemical, it was originally a prob(7) but since the chemical is now metabolized during every life() I reduced it to prob(5) so in fact I buffed it! The prob(33) that was in the proc was for the adjustToxLoss, all I did was remove the probably and randomness, instead of 33% of rand(5,25) once every two life we now get 2.5 every life() (0.33*0.5*(5+25)/2 = 2.5)

Iamgoofball replied:
Ah.

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+
+datum/reagent/toxin/amanitin/on_mob_delete(var/mob/living/M as mob)
+	M.adjustToxLoss(current_cycle*3*REM)
```
Take the *REM out of this.

phil235 replied:
REM was already there, it's used for nearly all reagent damage, plus it's equal to 1 by default.

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -327,6 +324,289 @@ datum/reagent/toxin/staminatoxin/on_mob_life(mob/living/carbon/M)
+	toxpwr = 0.75
+
+datum/reagent/toxin/cyanide/on_mob_life(var/mob/living/M as mob)
+	if(prob(5))
```
Why is this prob(5)?

Iamgoofball commented:
Overall, 7/10. Fix those marked issues and we're good to go.

Iamgoofball commented:
Also, I intentionally had random_color_list global so that we could use it in the future for other things.

optimumtact commented:
33 percent chance to stun was shit goof

Iamgoofball commented:
Then lower the stun chance and don't wrecking ball the rest?

phil235 committed:
Fix reagents not using current_cycle var for their stuff.
Fixes current_cycle being incremented twice for certain reagents.
Fixes flash never working for /datum/effect/effect/system/reagents_explosion because of a typo.
Readding nitroglycerin reagent, it can now be stabilized with stabilizing agent and can explodes when exposed to 474K (like blackpowder).
Readjusted black powder explosion: no more flash (like nitroglycerin), tweaked the values a bit to not nerf blackpowder explosion.
Adding changelog.

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Chems

Conflicts:
	code/modules/mob/living/carbon/alien/humanoid/humanoid.dm
	code/modules/mob/living/carbon/carbon.dm

phil235 commented:
Should be ready to go.

phil235 commented:
Good job Travis...
